interfaces/cython/cantera/cti2yaml.py

```diff
@@ -1632,9 +1631,23 @@ def convert(filename=None, output_name=None, text=None):
         else:
             # Error in cti2yaml or elsewhere
             traceback.print_exc()
-
         sys.exit(4)
 
+    # get file description from header block
+    if header:
+        description = []
+        for line in text.splitlines():
+            # only consider comments in the initial header block
+            # comments start with '#'; there may be empty leading lines
+            if line.startswith("#") or not description:
+                description.append(line[1:].rstrip())
```

Comment:
This won't be able to capture descriptions where multiple paragraphs are separated by blank lines, like:

```python
# Comment

# Comment
```

I'm also not 100% sure this won't give an `IndexError` for files that start with blank lines (I don't remember if `splitlines()` retains the trailing `\n` or not). I _think_ something like this might be more clear:

```python
leading_lines = True
for line in text.splitlines():
    if not line.strip() and leading_lines:
        continue
    else:
        leading_lines = False
    if line.strip() and not line.startswith("#"):
        break
    description.append(line[1:].rstrip())
```

Reply:
It does work with leading blank lines. Regarding concatenation of multiple comment blocks, not permitting this is a deliberate choice. `data/inputs/lithium_ion_battery.cti` is again a good example where you'd get lines that you may not want

Reply:
> It does work with leading blank lines.

Thanks! I wonder if the leading lines should be dropped though? I guess this will only collect one blank line if there are several of them?

> Regarding concatenation of multiple comment blocks, not permitting this is a deliberate choice.

OK, I think the other choice is the better one :smile: I'm not sure there's a good way to decide here... I guess I come down on the side of capturing more information since it doesn't affect the syntax of the file, rather than losing that information.

Reply:
I ended up with the following:
```python
    # get file description from header block
    description = []
    for line in text.splitlines():
        # only consider comments in the initial contiguous comment block
        # comments start with '#'; there may be empty leading lines
        if line.startswith("#") or not description:
            description.append(line[1:].rstrip())
        else:
            break
    description = textwrap.dedent("\n".join(description).strip("\n"))
```
which I believe is short and does exactly what we want.

I ultimately think that only using the first comment block makes the most sense. Users can very easily update the CTI input to modify the description (presumably they are more familiar with the legacy CTI syntax than the new YAML format).